assignments/models.py

```diff
@@ -59,6 +67,70 @@ class Meta:
         verbose_name = _('open text task')
         verbose_name_plural = _('open text tasks')
 
+
+class BudgetTextTask(models.Model):
+    """
+    Budget text task. Budget si related to section and consists of one or more Budgeting targets
+    """
+    UNIT_HA = 0
+    UNIT_PCS = 1
+    UNIT_CHOICES = (
+        (UNIT_HA, _('ha')),
+        (UNIT_PCS, _('pcs'))
+    )
+
+    section = models.ForeignKey(Section, related_name='budget_text_tasks')
+    name = models.CharField(_('name'), max_length=256)
+    unit = models.IntegerField(choices=UNIT_CHOICES, default=UNIT_HA)
+
+    class Meta:
+        verbose_name = _('budget text task')
+        verbose_name_plural = _('budget text tasks')
+
     def __str__(self):
-        return 'Question: {}'.format(self.question)
+        return self.name
+
+
+class BudgetTextTarget(models.Model):
+    """
+    Budget text task targets.
+    """
 
+    task = models.ForeignKey(BudgetTextTask, related_name='budget_targets')
+    name = models.CharField(_('name'), max_length=256)
+    unit_price = models.DecimalField(_('price'), max_digits=10, decimal_places=2, default=0)
+    reference_amount = models.DecimalField(_('reference amount'), max_digits=10, decimal_places=2, default=0)
+    range = FloatRangeField(null=True, blank=True)
+
+    class Meta:
+        verbose_name = _('budget text target')
+        verbose_name_plural = _('budget text targets')
+
+    def __str__(self):
+        return self.name
+
+
+class MapObject(models.Model):
```

Comment:
If you think of it, how is this different from `BudgetTarget`? I think if `BudgetTargets` had icon field, you would already have all the necessary information in a `BudgetTarget` to be able to drop them on a map if you wanted.